Servers/UI/OJS.Servers.Ui/ClientApp/src/components/modal/ContestModal.tsx

```diff
@@ -1,42 +1,42 @@
-import React, { useCallback } from 'react';
+import React, { useCallback, useState } from 'react';
 import { useNavigate } from 'react-router-dom';
 import Box from '@mui/material/Box';
 import Modal from '@mui/material/Modal';
 import Typography from '@mui/material/Typography';
 
-import { IContestModal } from '../../common/types';
+import { IContestModalInfoType } from '../../common/types';
 import { useAppUrls } from '../../hooks/use-app-urls';
 import { useCurrentContest } from '../../hooks/use-current-contest';
 import Button, { ButtonSize, ButtonType } from '../guidelines/buttons/Button';
 
 import styles from './ContestModal.module.scss';
 
 interface IContestModalProps {
-    contest: IContestModal;
-    isShowing: boolean;
-    toggle: () => void;
+    contest: IContestModalInfoType;
 }
 
-const ContestModal = ({ contest, isShowing, toggle }: IContestModalProps) => {
+const defaultState = { state: { isShowing: true } };
+
+const ContestModal = ({ contest }: IContestModalProps) => {
+    const [ isShowing, setIsShowing ] = useState<boolean>(defaultState.state.isShowing);
     const navigate = useNavigate();
     const { actions: { setIsUserParticipant } } = useCurrentContest();
     const { getHomePageUrl } = useAppUrls();
 
     const startContestAndHideModal = useCallback(
         () => {
             setIsUserParticipant(true);
-
-            toggle();
+            setIsShowing(false);
         },
-        [ setIsUserParticipant, toggle ],
+        [ setIsUserParticipant ],
     );
 
     const toggleAndRedirectToHomePage = useCallback(
         () => {
-            toggle();
+            setIsShowing(false);
             navigate(getHomePageUrl());
```

Comment:
Should the user be navigated to home page or back to contest page when closing the modal? I think It makes more sense to go back to contests page since this is the page from where the user opened the modal.